OS resource issues with recent changes (0.4.9)

Summary:

millw (since ~ 1 day ago) appears to:
- leave too many files open
- use too many native threads

---

I'm currently experimenting in CI to try and fix things, but I believe the issue is a recent change in `millw`.

This works fine:

```
# Download mill
RUN curl --fail -L -o /usr/local/bin/mill https://github.com/lihaoyi/mill/releases/download/0.10.12/0.10.12 && \
    chmod 755 /usr/local/bin/mill

...

a bunch of compilation + more Mill tasks
...
```
 

This has been having issues, since yesterday:

```
# Download millw
RUN curl --fail -L -o /usr/local/bin/mill https://raw.githubusercontent.com/lefou/millw/main/millw && \
    chmod 755 /usr/local/bin/mill
...
mill --version
mill __.compile
```

Even just invoking `--version` gets issues. The issues when using millw are like:

```
/usr/local/bin/mill: fork: retrty: Resource temporarily unavailable
/usr/local/bin/mill: fork: retrty: No child processes 
Caused by: java.io.IOException: Cannot run program "/usr/lib/jvm/java-17-amazon-corretto/bin/java": error=11, Resource temporarily unavailable
Exception in thread "main" java.lang.RuntimeException: java.io.IOException: Cannot run program "/usr/lib/jvm/java-17-amazon-corretto/bin/java": error=11, Resource temporarily unavailable
```

```
-       15:20:03  Error: Could not create the Java Virtual Machine.
-       15:20:03  Error: A fatal exception has occurred. Program will exit.
-       15:20:03  Could not flush log: stdout (Broken pipe (32))
-       15:20:03  Error: A JNI error has occurred, please check your installation and try again
-       15:20:03  Exception in thread "main" java.lang.OutOfMemoryError: unable to create native thread: possibly out of memory or process/resource limits reached
```
 
On some runs, too many files are left open - which causes issues for future Jenkins jobs running on the same swarm agents. They can fail with some OS tools, python scripts, whatever - anything that wants to read files:

```
10:54:07  ImportError: libk5crypto.so.3: cannot open shared object file: Too many open files in system
10:54:07  Traceback (most recent call last):
10:54:07    File "/usr/libexec/urlgrabber-ext-down", line 22, in <module>
...
10:54:07  ImportError: librt.so.1: cannot open shared object file: Too many open files in system
```

Mostly I've been blowing away the EC2 instance each time to clear things (trying to avoid blowing up my own PC's resources)

--- a/millw
+++ b/millw
@@ -114,6 +114,13 @@ try_to_use_system_mill() {
     return 0
   fi
 
+  SYSTEM_MILL_FIRST_TWO_BYTES=$(head --bytes=2 "${MILL_IN_PATH}")
+  if [ "${SYSTEM_MILL_FIRST_TWO_BYTES}" = "#!" ]; then
+	  # MILL_IN_PATH is (very likely) a shell script and not the mill
+	  # executable, ignore it.
+	  return 0
+  fi
+
   SYSTEM_MILL_PATH=$(readlink -e "${MILL_IN_PATH}")
   SYSTEM_MILL_SIZE=$(stat --format=%s "${SYSTEM_MILL_PATH}")
   SYSTEM_MILL_MTIME=$(stat --format=%y "${SYSTEM_MILL_PATH}")

